File: lab01/p04/main.cpp
#include <iostream>
#include <string>
#include <cctype>
#include <algorithm>

using namespace std;

int main(){
    for(string line; getline(cin, line);){
        string buff; 
        for(auto c : line){
            if(isalpha(c)){ //omit puntuation, space and other marks
                buff += tolower(c);
            }
        }

        // value semantics 
        string temp = buff;
        reverse(temp.begin(), temp.end());

        cout << (buff == temp ? "palindrome\n" : "not a palindrome\n");
    }
}

File: lab01/p04/Main.java
import java.util.*;

public class Main{
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        while(scanner.hasNextLine()){
            String line = scanner.nextLine();

            StringBuilder buff = new StringBuilder();
            for(char c : line.toCharArray()){
                if(Character.isLetter(c)){
                    buff.append(Character.toLowerCase(c));
                }
            }

            StringBuilder temp = buff; //reference semantics 
            //StringBuilder temp = new StringBuilder(buff);
            temp.reverse();
            
            System.out.println(buff.toString().equals(temp.toString()) ? "palindrome" : "not a palindrome");
        }
    }
}


